sql/hive/src/test/scala/org/apache/spark/sql/hive/client/HiveClientSuite.scala

```diff
@@ -74,23 +86,23 @@ class HiveClientSuite(version: String)
   test(s"getPartitionsByFilter returns all partitions when $tryDirectSqlKey=false") {
     val client = init(false)
     val filteredPartitions = client.getPartitionsByFilter(client.getTable("default", "test"),
-      Seq(parseExpression("ds=20170101")))
+      Seq(EqualTo(attr("ds"), Literal(20170101, IntegerType))))
```

Comment:
we can import `import org.apache.spark.sql.catalyst.dsl.expressions._` to simplify expression creation